hawkbit-rest/hawkbit-mgmt-resource/src/test/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtTargetTypeResourceTest.java

```diff
@@ -531,6 +535,36 @@ void searchTargetTypeRsql() throws Exception {
                 .andExpect(jsonPath("content[1].name", equalTo("TestName1234")));
     }
 
+    @Test
+    @WithUser(principal = TEST_USER, allSpPermissions = true)
+    @Description("Verifies quota enforcement for /rest/v1/targettypes/{ID}/compatibledistributionsettypes POST requests.")
+    void assignDistributionSetTypeToTargetTypeUntilQuotaExceeded() throws Exception {
+        String typeName = "TestTypeQuota";
+        final TargetType testType = createTestTargetTypeInDB(typeName);
+
+        // create distribution set types
+        final int maxDistributionSetTypes = quotaManagement.getMaxDistributionSetTypesPerTargetType();
+        final List<Long> dsTypeIds = Lists.newArrayList();
+        for (int i = 0; i < maxDistributionSetTypes + 1; ++i) {
+            final DistributionSetTypeCreate dsCreate = entityFactory.distributionSetType().create().name("dsType_" + i)
+                    .description("dsType_" + i).colour("#000000").key("dsType_" + i);
```

Comment:
We could use the testDataFactory instead

Reply:
👍  - adapted.